usl_models/usl_models/flood_ml/model.py

```diff
@@ -37,21 +32,41 @@ class Input(TypedDict):
 
     def __init__(
         self,
-        model_params: FloodModelParams,
+        params: FloodModelParams | None,
         spatial_dims: tuple[int, int] = (constants.MAP_HEIGHT, constants.MAP_WIDTH),
     ):
         """Creates the flood model.
 
         Args:
-            model_params: A dictionary of configurable model parameters.
+            params: A dictionary of configurable model parameters.
             spatial_dims: Tuple of spatial height and width input dimensions.
                 Needed for defining input shapes. This is an optional arg that
                 can be changed (primarily for testing/debugging).
+            artifact_uri: Optional artifact URI to load model weights from.
         """
-        self._model_params = model_params
+        self._model_params = params or model_params.default_params()
         self._spatial_dims = spatial_dims
         self._model = self._build_model()
 
+    @classmethod
+    def from_checkpoint(cls, artifact_uri: str, **kwargs) -> "FloodModel":
+        """Loads the model from a checkpoint URI.
+
+        We load weights only to keep custom methods (e.g. `call_n`) intact.
+        This only works if the model architecture is identical to the architecure
+        used during export.
+        Ideally, we would load the entire Keras model and use that directly to allow
+        loading different architectures within the same wrapper class.
+        Unfortunately, `call_n` is not trivially serializeable in its current state.
+        """
+        model = cls(**kwargs)
+        loaded_model = keras.models.load_model(os.path.join(artifact_uri, "model"))
```

Comment:
Turns out neither of these approaches work. PurePosixPath will collapse `//` -> `/`, and urljoin doesn't recognize `gs://` as a valid URL prefix.
Let's just expect `/model` in the input URI.

Reply:
is that because `artificat_uri` has a trailing `/` ? Can do `artifact_uri.rstrip('/')`

Reply:
`rstrip` would work. But I think it's better for this to be handled at the callsite in case other file formats are supported later or if the model export logic changes.